tests/hw-accel/nfd/features/tests/features-test.go

```diff
@@ -240,7 +242,88 @@ var _ = Describe("NFD", Ordered, func() {
 			}
 
 		})
+		It("Check if NFD Feature is deleted when node is deleted", reportxml.ID("68304"), func() {
+			skipIfConfigNotSet(nfdConfig)
+
+			By("Get initial node features")
+			initialNodeFeatures, err := get.GetNodeFeatures(APIClient)
+			glog.V(ts.LogLevel).Infof("Initial node features: %v", initialNodeFeatures)
+			Expect(err).NotTo(HaveOccurred(), "Failed to get initial node features")
+
+			initialFeatureCount := len(initialNodeFeatures)
+			glog.V(ts.LogLevel).Infof("Initial node features count: %d", initialFeatureCount)
+
+			By("Get current nodes")
+			nodes, err := Nodes.List(APIClient, metav1.ListOptions{
+				LabelSelector: labels.Set(GeneralConfig.WorkerLabelMap).String(),
+			})
+			Expect(err).NotTo(HaveOccurred(), "Failed to get nodes")
+			glog.V(ts.LogLevel).Infof("Current nodes count: %d", len(nodes))
+			if len(nodes) == 0 {
+				Skip("No worker nodes found to test node deletion")
+			}
+
+			// Select the first worker node for deletion
+			nodeToDelete := nodes[0]
+			glog.V(ts.LogLevel).Infof("Selected node for deletion: %s", nodeToDelete.Object.Name)
+
+			By("Delete the selected node")
+			err = nodeToDelete.Delete()
```

Comment:
is the node put back on the system ?
Asking because this is a potential destructive tests which, if there is some other test running, of other operators, might cause problem.
I would suggest adding a flag to run destructive tests, just in case